src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultValidationMetadataProvider.cs

```diff
@@ -12,15 +11,24 @@ namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
     /// </summary>
     public class DefaultValidationMetadataProvider : IValidationMetadataProvider
     {
+        private readonly ValidationErrorMetadata _validationErrorMetadata;
+
+        public DefaultValidationMetadataProvider(ValidationErrorMetadata validationErrorMetadata)
+        {
+            _validationErrorMetadata = validationErrorMetadata;
+        }
+
         /// <inheritdoc />
         public void GetValidationMetadata([NotNull] ValidationMetadataProviderContext context)
         {
+            context.ValidationMetadata.ValidationErrorMetadata = _validationErrorMetadata;
```

Comment:
Main oddity here is that a later `IValidationMetadataProvider` that wants to change a single `Func<string>` must first replace the containing `ValidationErrorMetadata` with a new instance. Thoughts on the following options?

1. Lots of docs about per-metadata customization of this property.
2. Create a new `ValidationErrorMetadata` instance here, avoiding the issue but creating a fair number of new objects that will be identical in most cases. (This option should be very safe: Hard to imagine a configuration delegate running after the first invocation of this method. Just not sure we need that level of safety here.)